content/concepts/did-ddo.md

```diff
@@ -1,173 +1,607 @@
-         "values":[
-            "0x2222",
-            "0x333"
-         ]
+        "type": "address",
+        "values": ["0x2222", "0x333"]
       }
-   ]
+    ]
+  }
 }
 ```
 
-For future usage, we can extend that with different credentials types. Example:
+### DDO Checksum
+
+In order to ensure the integrity of the DDO, a checksum is computed for each DDO:
+
+```js
+const checksum = sha256(JSON.stringify(ddo))
+```
+
+The checksum hash is used when publishing/updating metadata using the `setMetaData` function in the ERC721 contract, and is stored in the event generated by the ERC721 contract:
+
+```solidity
+event MetadataCreated(
+  address indexed createdBy,
+  uint8 state,
+  string decryptorUrl,
+  bytes flags,
+  bytes data,
+  bytes metaDataHash,
+  uint256 timestamp,
+  uint256 blockNumber
+);
+
+event MetadataUpdated(
+  address indexed updatedBy,
+  uint8 state,
+  string decryptorUrl,
+  bytes flags,
+  bytes data,
+  bytes metaDataHash,
+  uint256 timestamp,
+  uint256 blockNumber
+);
+```
+
+_Aquarius_ should always verify the checksum after data is decrypted via a _Provider_ API call.
+
+### State
+
+Each asset has a state, which is held by the NFT contract. The possible states are:
+
+| State   | Description                     |
+| ------- | ------------------------------- |
+| **`0`** | Active.                         |
+| **`1`** | End-of-life.                    |
+| **`2`** | Deprecated (by another asset).  |
+| **`3`** | Revoked by publisher.           |
+| **`4`** | Ordering is temporary disabled. |
+
+## Aquarius Enhanced DDO Response
+
+The following fields are added by _Aquarius_ in its DDO response for convenience reasons, where an asset returned by _Aquarius_ inherits the DDO fields stored on-chain.
+
+These additional fields are never stored on-chain, and are never taken into consideration when [hashing the DDO](#ddo-hash).
+
+### NFT
+
+The `nft` object contains information about the ERC721 NFT contract which represents the intellectual property of the publisher.
+
+| Attribute     | Type                   | Description                                                               |
+| ------------- | ---------------------- | ------------------------------------------------------------------------- |
+| **`address`** | `string`               | Contract address of the deployed ERC721 NFT contract.                     |
+| **`name`**    | `string`               | Name of NFT set in contract.                                              |
+| **`symbol`**  | `string`               | Symbol of NFT set in contract.                                            |
+| **`owner`**   | `string`               | ETH account address of the NFT owner.                                     |
+| **`state`**   | `number`               | State of the asset reflecting the NFT contract value. See [State](#state) |
+| **`created`** | `ISO date/time string` | Contains the date of NFT creation.                                        |
```

Comment:
this also needs the `tokenURI` as otherwise frontend needs to do a contract call for it as I realized when dealing with https://github.com/oceanprotocol/market/issues/944